### ISSUE TITLE
FFMPEG Producer fails to load USB-Inputs

### ISSUE THREAD

**Issue by Examed** [CasparCG/server#1101]
### Expected behaviour
CasparCG 2.1 and earlier builds are loading USB inputs like the Logitech Brio or C920 as Imput

### Current behaviour
When trying to load an USB input it is throwing the error below. 

----

### Steps to reproduce
1. PLAY 1-10 "dshow://video=USB Device NAME"

### LOG
```
[2018-11-01 16:03:05.647] [info]    Received message from 127.0.0.1: PLAY 1-10 "dshow://video=Logitech BRIO" --  -framerate 30 -video_size 1920x1080 -copyts\r\n
[2018-11-01 16:03:05.648] [info]    Sent message to 127.0.0.1:202 PLAY OK\r\n
[2018-11-01 16:03:05.648] [error]   Exception: C:\Program Files (x86)\Jenkins\workspace\casparcg-server-dep\master\src\modules\ffmpeg\producer\av_input.cpp(125): Throw in function void __cdecl caspar::ffmpeg::Input::reset(void)
[2018-11-01 16:03:05.648] [error]   Dynamic exception type: class boost::exception_detail::clone_impl<struct caspar::ffmpeg::ffmpeg_error_t>
[2018-11-01 16:03:05.648] [error]   [struct boost::errinfo_api_function_ * __ptr64] = avformat_open_input(&ic, filename_.c_str(), nullptr, &options)
[2018-11-01 16:03:05.648] [error]   [struct boost::errinfo_errno_ * __ptr64] = 1330794744, "Unknown error"
[2018-11-01 16:03:05.648] [error]   [struct caspar::tag_stacktrace_info * __ptr64] =  0# 0x00007FF729D2981E in casparcg
[2018-11-01 16:03:05.648] [error]    1# 0x00007FF729D58BC0 in casparcg
[2018-11-01 16:03:05.648] [error]    2# 0x00007FF729F31BFA in casparcg
[2018-11-01 16:03:05.648] [error]    3# 0x00007FF729F2BE37 in casparcg
[2018-11-01 16:03:05.648] [error]    4# 0x00007FF729F25F62 in casparcg
[2018-11-01 16:03:05.648] [error]    5# 0x00007FF72A0A5F33 in casparcg
[2018-11-01 16:03:05.648] [error]    6# o_ceil in ucrtbase
[2018-11-01 16:03:05.648] [error]    7# BaseThreadInitThunk in KERNEL32
[2018-11-01 16:03:05.648] [error]    8# RtlUserThreadStart in ntdll
[2018-11-01 16:03:05.648] [error]
[2018-11-01 16:03:05.648] [error]
[2018-11-01 16:03:05.648] [error]    0# 0x00007FF729D2981E in casparcg
[2018-11-01 16:03:05.648] [error]    1# 0x00007FF729D2914F in casparcg
[2018-11-01 16:03:05.648] [error]    2# 0x00007FF72A247C47 in casparcg
[2018-11-01 16:03:05.648] [error]    3# _C_specific_handler in VCRUNTIME140
[2018-11-01 16:03:05.648] [error]    4# _FrameUnwindFilter in VCRUNTIME140
[2018-11-01 16:03:05.648] [error]    5# RtlCaptureContext in ntdll
[2018-11-01 16:03:05.648] [error]    6# 0x00007FF729F25F62 in casparcg
[2018-11-01 16:03:05.648] [error]    7# 0x00007FF72A0A5F33 in casparcg
[2018-11-01 16:03:05.648] [error]    8# o_ceil in ucrtbase
[2018-11-01 16:03:05.648] [error]    9# BaseThreadInitThunk in KERNEL32
[2018-11-01 16:03:05.648] [error]   10# RtlUserThreadStart in ntdll
```

### Environment
* Server version: [2.2][casparcg-server-727542fc4d288c07ecebfda2e270ce3f55fc2329-windows.zip]
* Operating system: [e.g. Windows 10]

### Startup Log
```
[2018-11-01 15:59:45.637] [info]    ############################################################################
[2018-11-01 15:59:45.638] [info]    CasparCG Server is distributed by the Swedish Broadcasting Corporation (SVT)
[2018-11-01 15:59:45.638] [info]    under the GNU General Public License GPLv3 or higher.
[2018-11-01 15:59:45.638] [info]    Please see LICENSE.TXT for details.
[2018-11-01 15:59:45.638] [info]    http://www.casparcg.com/
[2018-11-01 15:59:45.638] [info]    ############################################################################
[2018-11-01 15:59:45.638] [info]    Starting CasparCG Video and Graphics Playout Server 2.2.0 727542fc Beta 9
[2018-11-01 15:59:45.678] [info]    Initialized ffmpeg module.
[2018-11-01 15:59:45.678] [info]    Initialized oal module.
[2018-11-01 15:59:45.678] [info]    Initialized decklink module.
[2018-11-01 15:59:45.678] [info]    Initialized screen module.
[2018-11-01 15:59:45.701] [info]    Initialized html module.
[2018-11-01 15:59:45.845] [info]    Initialized flash module.
[2018-11-01 15:59:45.846] [info]    Initialized newtek module.
[2018-11-01 15:59:45.846] [info]    Initialized bluefish module.
[2018-11-01 15:59:45.846] [info]    Initialized image module.
[2018-11-01 15:59:45.846] [info]    "C:/CasparCG Server - Beta\casparcg.config":
[2018-11-01 15:59:45.846] [info]    -----------------------------------------
[2018-11-01 15:59:45.846] [info]    <?xml version="1.0" encoding="utf-8"?>
[2018-11-01 15:59:45.846] [info]    <configuration>
[2018-11-01 15:59:45.846] [info]       <paths>
[2018-11-01 15:59:45.846] [info]          <media-path>media/</media-path>
[2018-11-01 15:59:45.846] [info]          <log-path>log/</log-path>
[2018-11-01 15:59:45.846] [info]          <data-path>data/</data-path>
[2018-11-01 15:59:45.846] [info]          <template-path>template/</template-path>
[2018-11-01 15:59:45.846] [info]          <font-path>font/</font-path>
[2018-11-01 15:59:45.846] [info]       </paths>
[2018-11-01 15:59:45.846] [info]       <lock-clear-phrase>secret</lock-clear-phrase>
[2018-11-01 15:59:45.846] [info]       <channels>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <consumers/>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <consumers/>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <channel-layout>stereo</channel-layout>
[2018-11-01 15:59:45.846] [info]             <consumers>
[2018-11-01 15:59:45.846] [info]                <decklink>
[2018-11-01 15:59:45.846] [info]                   <device>1</device>
[2018-11-01 15:59:45.846] [info]                   <embedded-audio>true</embedded-audio>
[2018-11-01 15:59:45.846] [info]                   <latency>normal</latency>
[2018-11-01 15:59:45.846] [info]                   <buffer-depth>8</buffer-depth>
[2018-11-01 15:59:45.846] [info]                </decklink>
[2018-11-01 15:59:45.846] [info]             </consumers>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <consumers/>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <consumers/>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <consumers>
[2018-11-01 15:59:45.846] [info]                <system-audio>true</system-audio>
[2018-11-01 15:59:45.846] [info]             </consumers>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]          <channel>
[2018-11-01 15:59:45.846] [info]             <video-mode>1080p3000</video-mode>
[2018-11-01 15:59:45.846] [info]             <consumers/>
[2018-11-01 15:59:45.846] [info]          </channel>
[2018-11-01 15:59:45.846] [info]       </channels>
[2018-11-01 15:59:45.846] [info]       <controllers>
[2018-11-01 15:59:45.846] [info]          <tcp>
[2018-11-01 15:59:45.846] [info]             <port>5250</port>
[2018-11-01 15:59:45.846] [info]             <protocol>AMCP</protocol>
[2018-11-01 15:59:45.846] [info]          </tcp>
[2018-11-01 15:59:45.846] [info]       </controllers>
[2018-11-01 15:59:45.846] [info]       <amcp>
[2018-11-01 15:59:45.846] [info]          <media-server>
[2018-11-01 15:59:45.846] [info]             <host>localhost</host>
[2018-11-01 15:59:45.846] [info]             <port>8000</port>
[2018-11-01 15:59:45.846] [info]          </media-server>
[2018-11-01 15:59:45.846] [info]       </amcp>
[2018-11-01 15:59:45.846] [info]    </configuration>
[2018-11-01 15:59:45.846] [info]    -----------------------------------------
[2018-11-01 15:59:46.118] [info]    Initializing OpenGL Device.
[2018-11-01 15:59:46.119] [info]    Initialized OpenGL 4.5.13492 Core Profile Context 22.19.677.257 ATI Technologies Inc.
[2018-11-01 15:59:46.188] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 1
[2018-11-01 15:59:46.188] [info]    video_channel[1|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.189] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 2
[2018-11-01 15:59:46.189] [info]    video_channel[2|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.189] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 3
[2018-11-01 15:59:46.189] [info]    video_channel[3|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.190] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 4
[2018-11-01 15:59:46.190] [info]    video_channel[4|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.190] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 5
[2018-11-01 15:59:46.191] [info]    video_channel[5|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.191] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 6
[2018-11-01 15:59:46.191] [info]    video_channel[6|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.191] [info]    Initialized OpenGL Accelerated GPU Image Mixer for channel 7
[2018-11-01 15:59:46.191] [info]    video_channel[7|1080p3000] Successfully Initialized.
[2018-11-01 15:59:46.216] [warning] Device does not support video-format: 1080p30
[2018-11-01 15:59:46.217] [info]    DeckLink Mini Monitor [3-1|1080p3000] Enabled embedded-audio.
[2018-11-01 15:59:46.217] [info]    DeckLink Mini Monitor [3-1|1080p3000] Disabled low-latency mode.
[2018-11-01 15:59:46.217] [error]   DeckLink Mini Monitor [3-1|1080p3000] Failed to enable external keyer.
[2018-11-01 15:59:46.242] [info]    DeckLink Mini Monitor [3-1|1080p3000] Initialized.
[2018-11-01 15:59:46.275] [info]    oal[6|1080p3000] Initialized.
[2018-11-01 15:59:46.275] [info]    Initialized channels.
[2018-11-01 15:59:46.276] [info]    Initialized controllers.
[2018-11-01 15:59:46.276] [info]    Initialized osc.
```

**Comment by Julusian** [comment]
It looks like the `dshow://` protocol was something that was a bit custom for 2.0.7 to help caspar determine it is a device rather than a file/stream, and that logic doesn't appear to be in replicated in 2.2 currently

**Comment by Julusian** [comment]
This is fixed in the latest 2.2.x build.
